### ISSUE TITLE
Wrong POD config used for wrong branches

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1495]
Wrong POD config used for wrong branches

https://ci-staging.eclipse.org/xtext/job/xtext-xtend/job/master/148/console
does not use the CBI.Jenkinsfile from master 

https://bugs.eclipse.org/bugs/show_bug.cgi?id=548573

**Comment by cdietrich** [comment]
another sample of random pick: https://ci-staging.eclipse.org/xtext/job/xtext-xtend/job/kt_issue1431/16/console

**Comment by Hejado** [comment]
See here: https://issues.jenkins-ci.org/browse/JENKINS-47759

While I have an issue with how this works, it seems we'll need to define a unique label for each pod template.

It would explain much of what we saw while testing this, but even when using unique labels, we did see strange behaviour. I assume, if you use the same label but wait long enought between two jobs (like we did with the cd_experiment*-branches, it will use the locally declared yaml. But if the job queue is full and Jenkins schedules builds much quicker (like it did when we started testing this), there might be some overlap.

So, in short: Hopefully using unique labels will work... Will provide PRs to test this shortly...

**Comment by cdietrich** [comment]
@kthoms @Hejado  can we consider this fixed?

**Comment by kthoms** [comment]
Seems that this works now.
